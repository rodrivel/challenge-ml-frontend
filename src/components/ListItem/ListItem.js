import React from 'react';
import PropTypes from 'prop-types';
import cx from './ListItem.module.scss';
import { Link } from 'react-router-dom';
import Skeleton from 'react-loading-skeleton';
import iconFreeShipping from '../../assets/ic_shipping@2x.png.png';
import { currencyFormat } from '../../helpers';


function ListItem(props) {
  const { itemData } = props;
  const { price } = itemData;
  return (
      <div className={cx.ListItem}>
        <div className={cx.ListItem__Wrapper}>
          <div className={cx.ListItem__ImgWrapper}>
            <Link to={`/items/${itemData.id}`} className={cx.ListItem__Link}>
              { itemData.picture 
                ? <img src={itemData.picture} alt={`Imagen ${itemData.title}`}/>
                : <Skeleton style={{ height: '180px'}}/>
              }              
            </Link>
          </div>
          <div className={cx.ListItem__MainInfo}>
            <Link to={`/items/${itemData.id}`} className={cx.ListItem__Link}>
                { itemData.price 
                  ? (
                    <div className={cx.ListItem__MainInfo__Price}>
                      <div className={cx.ListItem__MainInfo__Price__Block}>
                        <span>{price && price.currency === 'ARS' ? '$ ' : price.currency }{currencyFormat(price.amount)}</span>
                        { itemData.price.decimals > 0 ? <span className={cx.ListItem__MainInfo__Price__Decimals }>{ itemData.price.decimals }</span> : null }
                        { itemData.free_shipping && <span><img width={20} src={iconFreeShipping}/></span> }
                      </div>
                    </div>
                  )
                  : <Skeleton height={30} width={150} style={{ marginBottom: '20px'}}/>
                }
              
              { itemData.title 
                ? <div className={cx.ListItem__MainInfo__Title}>{itemData.title}</div>
                : <Skeleton style={{ marginTop: '8px', display: 'block'}} width={'100%'} count={2}/>
              }         
            </Link>
            <div className={cx.ListItem__SecondaryInfo}>
              { itemData && itemData.address?.state_name 
                  ? <div className={cx.ListItem__SecondaryInfo__StateName}>{itemData.address.state_name}</div>
                  : <Skeleton width={100}/>
                }
            </div>          
          </div>                   
        </div>
        <div className={cx.ListItem__Divider}></div>    
      </div>
  );
}

ListItem.propTypes = {
  itemData: PropTypes.object.isRequired  
}

export default ListItem;