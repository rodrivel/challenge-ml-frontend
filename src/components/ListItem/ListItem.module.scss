@import 'src/styles/base';

.ListItem {
  display: flex;
  flex-direction: column;
  background-color: #fff;
  
  .ListItem__Wrapper {    
    border-radius: $border-radius;
    display: flex;
    padding: 16px;
    text-align: left;    
  
    .ListItem__Link {
      text-decoration: none;    
    }
  
    .ListItem__ImgWrapper {
      width: 180px;
      height: 180px;
      border-radius: $border-radius;
      overflow: hidden;
      margin-right: 16px;
      flex-shrink: 0;
  
      img {
        width: 100%;
        height: 100%;
        object-fit: cover;
      }
    }
  
    .ListItem__MainInfo {
      flex-grow: 1;
      font-size: 1.125rem;
      padding: 32px 0;
  
      .ListItem__MainInfo__Price {
        font-size: 1.5rem;
        color: $primary-text;
  
        .ListItem__MainInfo__Price__Block {
          display: flex;
          align-items: center;
          line-height: 1;
          color: $primary-text;
          gap: 10px;

          span {
            line-height: 1;
          }

          .ListItem__MainInfo__Price__Decimals {
            font-size: 16px;
            font-weight: 300;
            align-self: flex-start;
            margin-left: -5px;
          }
        }

      }
  
      .ListItem__MainInfo__Title {
        margin-top: 32px;      
        line-height: 1.3;
        color: $secondary-text;
      }
    }
  
    .ListItem__SecondaryInfo {            
      flex-shrink: 0;
      padding: 32px;
      width: 250px;

      .ListItem__SecondaryInfo__StateName {
        color: $secondary-text;
        font-size: 12px;
        font-weight: 300;
      }
    }
    
  }

  .ListItem__Divider {
    margin: 0 16px;
    border-bottom: 1px solid $light-gray;
  }
}