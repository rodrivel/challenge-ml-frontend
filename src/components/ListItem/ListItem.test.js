import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

it('expect to render ListItem component', () => {
  const mockItemData = {
    id: 'MLAXXXXXXX',
    title: 'lorem ipsum',
  };
  expect(shallow(<ListItem itemData={mockItemData} />)).toMatchSnapshot();
});
