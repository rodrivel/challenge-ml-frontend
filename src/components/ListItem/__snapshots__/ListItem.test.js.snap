// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ListItem component 1`] = `ShallowWrapper {}`;
